src/python/pants/backend/adhoc/target_types.py

```diff
@@ -216,6 +217,26 @@ class AdhocToolWorkdirField(StringField):
     )
 
 
+class AdhocToolNamedCachesField(DictStringToStringField):
+    alias = "experimental_named_caches"
+    default = {}
+    help = help_text(
+        """
+        Named caches to construct for the execution.
+        See https://www.pantsbuild.org/docs/reference-global#named_caches_dir.
+
+        The keys of the mapping are the directory name to be created in the named caches dir.
+        The values are the name of the symlink (relative to the sandbox root) in the sandbox which
+        points to the subdirectory in the named caches dir
+
+        NOTE: The named caches MUST be handled with great care. Processes accessing the named caches
+        can be run in parallel, and can be cancelled at any point in their execution (and
+        potentially restarted). That means that _every_ operation modifying the contents of the cache
+        MUST be concurrency and cancellation safe.
+        """
+    )
```

Comment:
Josh, I'm surprised you as a visual person didn't include an example here. ;)

Maybe an example doesn't give much, but it helps paint a picture of the above sentence which is kind of self referential and potentially slightly confusing (not saying it needs re-working, just that the concept isn't immediately apparent)

Reply:
the dirs

Reply:
....I'm gonna put for now, just because we re-use the field in two targets and I don't wanna muck with the help for that. It's experimental, so meh?